### PR TITLE
Upload files API endpoint

### PR DESCRIPTION
**Pull request by jpodivin**
Partially implements #5 as I haven't mastered the clojure yet.

FastAPI upload requires python-multipart[0] which only seems to have src package in fedora, so it has to be installed from pypi. 

[0] https://fastapi.tiangolo.com/tutorial/request-files/#request-files

### CODE REVIEW

**Comment by FrostyX** [comment]
> Partially implements https://github.com/fedora-copr/log-detective-website/issues/5 as I haven't mastered the clojure yet.

Should I implement the frontend part or do you want to play with it?

**Comment by jpodivin** [comment]
@FrostyX If you are fine with implementing the frontend that would be great. I doubt my performance in that area.

**Comment by FrostyX** [comment]
I am working on this and I am close to finishing the frontend side.

**Comment by FrostyX** [comment]
I spent my Sunday on this :tired_face:, but I think it finally works.

I pushed directly to your branch @jpodivin, hopefully, you don't mind. I changed/removed part of your code, not because a "not invented here" syndrome on my part, I promise. It turned out it was much easier to not have `UploadProvider` and do everything on frontend.

The reason we have `CoprProvider`, `URLProvider`, etc is that CORS wouldn't allow Javascript to send HTTP requests to random websites (AFAIK). That's why we need our backend to act as a middleman. But in the case of file upload, it is easier to save the uploaded file to [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage#examples) than using the backend as a middleman.

**Comment by FrostyX** [comment]
Please try if it works for you. Also, the `python-multipart` probably won't be needed anymore.

**Comment by FrostyX** [comment]
Oops, I wanted to link this tutorial in my commit message
https://tech.toryanderson.com/2021/09/17/clojurescript-reagent-image-previewing-selector/
Well, at least linking it here for the record.

**Review by nikromen — COMMENTED** [review]
Cool! It is nice we can have the upload implemented directly in frontend instead of hopping between BE -> FE -> BE. Just some notes below

- I can see we have only spec file option for upload, but how would we deal with containers? Is it only a feature for mock builds and later on we will add an option for container builds also?
- There is no image in the upload tab - seems like image does not exist
Screenshot_20240115_123301

File: frontend/src/app/contribute.cljs [hunk 1]
```diff
@@ -58,40 +77,38 @@
...
-
-                     (reset! error-title nil)
-                     (reset! error-description nil))))))))
+                   (set-atoms data)))))))
+
+(defn fetch-logs-upload []
+  (let [data {:build_id_title "Upload"
+              :logs [{:name (.getItem js/localStorage "name")
+                      :content (.getItem js/localStorage "content")}]
+              :spec_file {:name "fake_spec_name.spec"
```
Comment (nikromen) [inline frontend/src/app/contribute.cljs:86]: can we return here `nil` instead of fake spec file? to avoid the same problem as #53

Comment (FrostyX) [reply frontend/src/app/contribute.cljs]: Nice, done :-)

**Comment by FrostyX** [comment]
> There is no image in the upload tab - seems like image does not exist

Good catch, I forgot to commit it.

> I can see we have only spec file option for upload, but how would we deal with containers? Is it only a feature for mock builds and later on we will add an option for container builds also?

Yeah, unfortunately, it's only for RPM build logs at the moment. Do you see it as a blocker?

**Review by nikromen — APPROVED** [review]
:rocket:

**Comment by nikromen** [comment]
> Yeah, unfortunately, it's only for RPM build logs at the moment. Do you see it as a blocker?

No, I think the URL upload for containers is good enough for containers. My concern was if users will know but I overlooked the `Upload RPM logs` in upload tab :see_no_evil: So I think it is clear that this is not about containers, my bad :D we can add upload for it later if somebody needs it

**Pull request merged**